api/app/signals/apps/email_integrations/tests/test_tasks.py

```diff
@@ -18,3 +19,50 @@ def test_send_mail_reporter_created(self, mocked_mail):
 
         tasks.send_mail_reporter(pk=signal.pk)
         mocked_mail.assert_called_once_with(signal=signal.pk)
+
+    @mock.patch('signals.apps.email_integrations.tasks.MailService.system_mail', autospec=True)
+    def test_send_mail_assigned_signal_departments(self, mocked_mail):
+        signal = SignalFactory.create()
+        department = DepartmentFactory.create()
+
+        user_with_notification = UserFactory.create()
+        user_with_notification.profile.notification_on_department_assignment = True
+        user_with_notification.profile.save()
+        user_with_notification.profile.departments.add(department)
+
+        user_without_notification = UserFactory.create()
```

Comment:
Not a blocker and not strictly necessary: I like it when tests are explicit - consider setting `notification_on_user_assignment` to `False`.

Reply:
Fixed this